### DISCUSSION TITLE
Provide a way to dynamically pre-fill API tokens

### DISCUSSION THREAD

**Discussion by miloandco** [scalar/scalar#1710] [Ideas]
Using scalar alongside your application, it would make sense to make API requests, as the current user authenticated in your application.

Therefore it would be great with a method to dynamically pre-fill e.g. Bearer tokens. A non integration/application specific solution to this, could be by simply add a token endpoint to the config. If this endpoint returns a token, scalar will pre fill the token field with this. 

The api token endpoint should be checked at runtime and not on build, as scalar should get the updated token, if the user authentication changes in your application.

---

Feature request as per encouragement by one of the developers on Discord.

**Comment by hanspagel** [comment]
Ideally it would be a callback for full flexibility. :)

**Reply by hanspagel** [reply]
I’ve created an issue here: https://github.com/scalar/scalar/issues/2010

**Comment by amritk** [comment]
Does updating the authentication config not work for this?
